src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = [];
+  const finalState = [];
+
+  newState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    for (const key in actions[i]) {
+      if (actions[i][key] === 'addProperties') {
```

Comment:
it's more convenient to use switch statement, and when you do:
3. [CODE STYLE]: switch/case should always have default case for error handling.